store/read_transaction.cpp

```diff
@@ -196,13 +196,22 @@ parse_result(sqlite::statement& stmt, const char* type_column,
     try {
         const model::test_result_type type =
             store::column_test_result_type(stmt, type_column);
-        if (type == model::test_result_passed) {
+        switch (type) {
+        case model::test_result_passed:
             if (stmt.column_type(stmt.column_id(reason_column)) !=
                 sqlite::type_null)
                 throw store::integrity_error("Result of type 'passed' has a "
                                              "non-NULL reason");
             return model::test_result(type);
-        } else {
+        /// Some test interfaces don't have to provide a reason when skipping a
+        /// test, e.g., Google Test.
+        case model::test_result_skipped:
```

Comment:
This change is quite suspicious, especially because it affects no tests in `store`.

I'd rather add a fake skip message to GoogleTest results for now and avoid touching this at all. You could make skip messages optional in a future step.

Reply:
> This change is quite suspicious, especially because it affects no tests in `store`.
> 
> I'd rather add a fake skip message to GoogleTest results for now and avoid touching this at all. You could make skip messages optional in a future step.

What's the advantage of including this in a separate commit?

Reply:
That you can add some tests for it which would make this change even bigger and harder to review.

Also note that this schema change effectively makes different versions of the database incompatible with each other: a database build with Kyua after this commit wouldn't be parseable by a previous version of Kyua. Which means the schema version should get a bump, and then a migration process put in place.